server/src/main/java/io/crate/role/Role.java

```diff
@@ -81,6 +167,18 @@ public boolean isUser() {
         return isUser;
     }
 
+    public Set<GrantedRole> parentRoles() {
+        return parentRoles;
+    }
+
+    public Set<String> parentRoleNames() {
+        Set<String> parentRoleNames = LinkedHashSet.newLinkedHashSet(parentRoles().size());
```

Comment:
Why order is important here?

Reply:
Really I'm not sure, in my head I would expect the privileges to be checked in the order they were granted but I didn't find mentioning something like that, and to achieve my thought would need to use LinkedHashSets as well for the `parentRoles`. I think it's better to use plain `HashSet`.